libxw/src/libxw/render.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Generic
+from typing import TypeVar
+
+from .puzzle import Cell
+from .puzzle import Puzzle
+
+
+OutputFormat = TypeVar("OutputFormat")
+
+
+class Renderer(ABC, Generic[OutputFormat]):
+    @abstractmethod
+    def render_puzzle(self, puzzle: Puzzle) -> OutputFormat:
+        ...
```

Comment:
This kinda crazy syntax is basically saying that each concrete `Renderer` subclass knows how to "render" puzzles into some specific `OutputFormat`.  I'm not sure we need all that flexibility (I can't imagine rendering to anything besides a string), but I think it's a useful mental model?